Software/Raspberry_Pi/pi/base_detect.py

```diff
@@ -0,0 +1,124 @@
+    detection_result = detector.detect(input_tensor)
+    
+    # Check detection results to send over
+    isObject = False
+    for detection_object in detection_result.detections:
+        bounding_box = detection_object.bounding_box
+        (x,y,w,h) = bounding_box.origin_x, bounding_box.origin_y, bounding_box.width, bounding_box.height
+        object_class = detection_object.categories[0].category_name
+        print(f'x,y,w,h,object_class: {x,y,w,h,object_class}')
+        ser.write(b'x,y,w,h,object_class: ')
+        x = max(x,0)        # Boundary boxes can be negative if top left coordinate can be inferred from visible regions
+        y = max(y,0)
+        ser.write(x)
+        ser.write(y)
+        ser.write(x)
+        ser.write(h)
+        ser.write(object_class.encode('utf-8'))
+        ser.write(b'\n')
+        line = ser.readline().decode('utf-8').rstrip()
+        print(line)
+        # time.sleep(1)
+        # time.sleep(1)
+        #line = ser.readline().decode('utf-8').rstrip()
+    # Send over the results
+
+    if visualize_image:
+
+        # Draw keypoints and edges on input image
+        image = utils.visualize(image, detection_result)
+
+        # Calculate the FPS
+        if counter % fps_avg_frame_count == 0:
+          end_time = time.time()
+          fps = fps_avg_frame_count / (end_time - start_time)
+          start_time = time.time()
+
+        # Show the FPS
+        fps_text = 'FPS = {:.1f}'.format(fps)
+        text_location = (left_margin, row_size)
+        cv2.putText(image, fps_text, text_location, cv2.FONT_HERSHEY_PLAIN,
+                    font_size, text_color, font_thickness)
+
+        # Stop the program if the ESC key is pressed.
+        if cv2.waitKey(1) == 27:
+          break
+        cv2.imshow('object_detector', image)
+
+    if cv2.waitKey(1)==27:
+        break
+
+  cap.release()
+  cv2.destroyAllWindows()
+
+def main():
+  parser = argparse.ArgumentParser(
+      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument('--model',help='Path of the object detection model.',required=False,default='efficientdet_lite0.tflite')
+  parser.add_argument('--cameraId', help='Id of camera.', required=False, type=int, default=0)
+  parser.add_argument('--frameWidth',help='Width of frame to capture from camera.',required=False,type=int,default=640)
+  parser.add_argument('--frameHeight',help='Height of frame to capture from camera.',required=False,type=int,default=480)
+  parser.add_argument('--numThreads',help='Number of CPU threads to run the model.',required=False,type=int,default=4)
+  parser.add_argument('--enableEdgeTPU',help='Whether to run the model on EdgeTPU.',action='store_true',required=False,default=False)
+  parser.add_argument('--visualize',help='Show camera',action='store_true',required=False,default=False)
+  args = parser.parse_args()
+  
+  run(args.model,args.cameraId,args.frameWidth,args.frameHeight,args.numThreads,args.enableEdgeTPU,args.visualize)
```

Comment:
Make sure these are set in a start-up script so that this program comes up on boot.